### PR TITLE
Fixes to storage documentation

### PR DESCRIPTION
**Pull request by TrekSoft**
Fixes the following problems with Storage documentation:

1) It wasn't clear you have to specify a globally unique combination of bucket name and friendly type name

2) The code samples were broken and not self contained

3) The s3 storage structure wasn't clear

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: android/storage.md [hunk 1]
```diff
@@ -32,13 +32,15 @@ See [Authentication](./authentication) for more information on how to get the `u
...
 
     `❯ Content (Images, audio, video, etc.)`
 
-4. The CLI walks you through the options to enable Auth (if not enabled previously), to name your S3 bucket, and to decide who should have access (select `Auth and guest users` and toggle all to select `create/update, read, and delete` access for both auth and guest users).
+4. The combination of friendly name and bucket name must be globally unique. If another S3 user has specified the same values for both of these as you, the amplify push step below will fail.
 
-5. Confirm that you have storage and auth set up.
+5. The CLI walks you through the options to enable Auth (if not enabled previously), to name your S3 bucket, and to decide who should have access (select `Auth and guest users` and toggle all to select `create/update, read, and delete` access for both auth and guest users).
+
+6. Confirm that you have storage and auth set up.
```
Comment (palpatim) [inline android/storage.md]: nit: I think our style is to capitalize category names

File: android/storage.md [hunk 2]
```diff
@@ -108,51 +110,73 @@ The TransferUtility checks the size of the file being uploaded and automatically
...
+        }
+        catch(Exception e) {
+            Log.e(TAG, e.getMessage());
+        }
 
         TransferObserver uploadObserver =
-            transferUtility.upload(
-                "public/s3Key.txt",
-                new File("/path/to/file/localFile.txt"));
+                transferUtility.upload(
```
Comment (palpatim) [inline android/storage.md]: please confirm that there are only spaces, no tabs, in the file

File: android/storage.md (hunk 1 above)
Comment (TrekSoft) [reply android/storage.md]: Done

File: android/storage.md (hunk 2 above)
Comment (TrekSoft) [reply android/storage.md]: Confirmed they are spaces not tabs

**Pull request merged**